examples/benchmarks/train_resnet_imagenet1k.py

```diff
@@ -0,0 +1,320 @@
+    transforms.RandomHorizontalFlip(),
+])
+train_dataset = ImageFolder(os.path.join(args.data_dir, 'train'), train_transforms)
+# Nifty function to instantiate a PyTorch DistributedSampler based on your hardware setup
+train_sampler = dist.get_sampler(train_dataset, drop_last=True, shuffle=True)
+train_dataloader = torch.utils.data.DataLoader(
+    train_dataset,
+    batch_size=args.train_batch_size,
+    num_workers=8,
+    pin_memory=True,
+    drop_last=True,
+    sampler=train_sampler,
+    collate_fn=pil_image_collate,
+    persistent_workers=True,  # Reduce overhead of creating new workers at the expense of using slightly more RAM
+)
+# DataSpec allows for on-gpu transformations, marginally relieving dataloader bottleneck
+train_dataspec = DataSpec(dataloader=train_dataloader,
+                          device_transforms=NormalizationFn(mean=IMAGENET_CHANNEL_MEAN, std=IMAGENET_CHANNEL_STD))
+logging.info('Built train dataloader')
+
+# Validation dataset
+logging.info('Build evaluation dataloader')
+eval_transforms = transforms.Compose([
+    transforms.Resize(args.eval_resize_size),
+    transforms.CenterCrop(args.eval_crop_size),
+])
+eval_dataset = ImageFolder(os.path.join(args.data_dir, 'val'), eval_transforms)
+# Nifty function to instantiate a PyTorch DistributedSampler based on your hardware setup,
+eval_sampler = dist.get_sampler(train_dataset, drop_last=False, shuffle=False)
+eval_dataloader = torch.utils.data.DataLoader(
+    eval_dataset,
+    batch_size=args.eval_batch_size,
+    num_workers=8,
+    pin_memory=True,
+    drop_last=False,
+    sampler=eval_sampler,
+    collate_fn=pil_image_collate,
+    persistent_workers=True,  # Reduce overhead of creating new workers at the expense of using slightly more RAM
+)
+eval_dataspec = DataSpec(dataloader=eval_dataloader,
+                         device_transforms=NormalizationFn(mean=IMAGENET_CHANNEL_MEAN, std=IMAGENET_CHANNEL_STD))
+logging.info('Built evaluation dataloader')
+
+# Instantiate torchvision ResNet model
+logging.info('Build Composer model')
+model_fn = getattr(resnet, args.model_name)
+model = model_fn(num_classes=1000, groups=1, width_per_group=64)
+
+
+# Specify model initialization
+def weight_init(w: torch.nn.Module):
+    if isinstance(w, torch.nn.Linear) or isinstance(w, torch.nn.Conv2d):
+        torch.nn.init.kaiming_normal_(w.weight)
+    if isinstance(w, torch.nn.BatchNorm2d):
+        w.weight.data = torch.rand(w.weight.data.shape)
+        w.bias.data = torch.zeros_like(w.bias.data)
+    # When using binary cross entropy, set the classification layer bias to -log(num_classes)
+    # to ensure the initial probabilities are approximately 1 / num_classes
+    if args.loss_name == 'binary_cross_entropy' and isinstance(w, torch.nn.Linear):
+        w.bias.data = torch.ones(w.bias.shape) * -torch.log(torch.tensor(w.bias.shape[0]))
+
+
+model.apply(weight_init)
+
+# Performance metrics to log outside of training loss
```

Comment:
```suggestion
# Performance metrics to log other than training loss
```
"outside of" kinda sounds like it's talking about scoping / where the logic runs